src/CodeGen_Internal.cpp

```diff
@@ -36,18 +37,20 @@ llvm::Type *llvm_type_of(LLVMContext *c, Halide::Type t) {
             return llvm::Type::getIntNTy(*c, t.bits());
         }
     } else {
-        llvm::Type *element_type = llvm_type_of(c, t.element_of());
-        return get_vector_type(element_type, t.lanes());
-    }
-}
-
-int get_vector_num_elements(llvm::Type *t) {
-    if (t->isVectorTy()) {
-        auto *vt = dyn_cast<llvm::FixedVectorType>(t);
-        internal_assert(vt) << "Called get_vector_num_elements on a scalable vector type\n";
-        return vt->getNumElements();
-    } else {
-        return 1;
+        llvm::Type *element_type = llvm_type_of(c, t.element_of(), 0);
+        bool scalable = false;
+        int lanes = t.lanes();
+        if (effective_vscale != 0) {
+            int total_bits = t.bits() * t.lanes();
+            scalable = ((total_bits % effective_vscale) == 0);
+            if (scalable) {
+                lanes /= effective_vscale;
+            } else {
+                debug(0) << "Failed to make scalable with bits " << t.bits() << " lanes " << t.lanes()
```

Comment:
Should this be internal_error?

Reply:
Done. Added comment to document the situation.